appscale/tools/agents/azure_agent.py

```diff
@@ -0,0 +1,689 @@
+    PARAM_SUBCR_ID,
+    PARAM_TENANT_ID,
+    PARAM_ZONE
+  )
+
+  # The admin username needed to create an Azure VM instance.
+  ADMIN_USERNAME = 'azureuser'
+
+  # The maximum number of seconds to sleep while waiting for
+  # Azure resources to get created.
+  SLEEP_TIME = 10
+
+  def assert_credentials_are_valid(self, parameters):
+    """ Contacts Azure with the given credentials to ensure that they are
+    valid. Gets an access token and a Credentials instance in order to be
+    able to access any resources.
+    Args:
+      parameters: A dict, containing all the parameters necessary to
+        authenticate this user with Azure.
+    Returns:
+      True, if the credentials were valid.
+      A list, of resource group names under the subscription.
+    Raises:
+      AgentConfigurationException: If an error is encountered during
+        authentication.
+    """
+    credentials = self.open_connection(parameters)
+    subscription_id = parameters[self.PARAM_SUBCR_ID]
+    try:
+      resource_client = ResourceManagementClient(credentials, subscription_id)
+      resource_groups = resource_client.resource_groups.list()
+      rg_names = []
+      for rg in resource_groups:
+        rg_names.append(rg.name)
+      return True, rg_names
+    except CloudError as error:
+      raise AgentConfigurationException("Unable to authenticate using the "
+        "credentials provided. Reason: {}".format(error.message))
+
+  def configure_instance_security(self, parameters):
+    """Configure and setup groups and storage accounts for the VMs spawned.
+    This method is called before starting virtual machines.
+    Args:
+      parameters: A dict containing values necessary to authenticate with the
+        underlying cloud.
+    Returns:
+      True, if the group and account were created successfully.
+      False, otherwise.
+    Raises:
+      AgentRuntimeException: If security features could not be successfully
+        configured in the underlying cloud.
+    """
+    credentials = self.open_connection(parameters)
+    resource_group = parameters[self.PARAM_RESOURCE_GROUP]
+    storage_account = parameters[self.PARAM_STORAGE_ACCOUNT]
+    zone = parameters[self.PARAM_ZONE]
+    subscription_id = parameters[self.PARAM_SUBCR_ID]
+
+    AppScaleLogger.log("Verifying that SSH key exists locally.")
+    keyname = parameters[self.PARAM_KEYNAME]
+    private_key = LocalState.LOCAL_APPSCALE_PATH + keyname
+    public_key = private_key + ".pub"
+
+    if os.path.exists(private_key) or os.path.exists(public_key):
+      raise AgentRuntimeException("SSH key already found locally - please "
+                                  "use a different keyname.")
+
+    LocalState.generate_rsa_key(keyname, parameters[self.PARAM_VERBOSE])
+
+    AppScaleLogger.log("Configuring network for machine/s under "
+                       "resource group '{0}' with storage account '{1}' "
+                       "in zone '{2}'".format(resource_group, storage_account, zone))
```

Comment:
If you're only using the storage account to print a message, then please remove it from the method docstring.